lockup: add vesting info, add graphs with schedule to the doc (#691)
near/docs: docs/tokens/lockup.md

3. Months 13-48: the remaining 75% of tokens are unlocked smoothly over each block of the remaining 36 months.
4. Months 48+: all tokens are unlocked

*See how NEAR tokens have been distributed and what lockups generally apply in [this post](https://near.org/blog/near-token-supply-and-distribution/).  If you want an easy way to check account balances or lockup details, you can lookup your account using [this tool](https://near.github.io/account-lookup).*

*See the FAQ at the end for questions*

> ## The Lockup Contract at Near

Reviewer: ```suggestion
## The Lockup Contract at NEAR
```

[Lockup](https://github.com/near/core-contracts/tree/master/lockup) is a special smart contract that ensures that the full or the partial amount is not transferable until it is supposed to be.

The lockups are implemented as a separate smart contract from your main account.

> Thus, if you have received tokens prior to [Phase 2](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:

Reviewer: ```suggestion
Thus, if you have received tokens prior to [Phase II](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:
```

1. A regular account (also called "Owner Account" in context of lockups), let's say `user.near` or `3e52c197feb13fa457dddd102f6af299a5b63465e324784b22aaa7544a7d55fb`;

> 2. A lockup contract, with name `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.

Reviewer: ```suggestion
2. A lockup contract, with a name like `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.
```

> The owner account is created first, either by following the NEAR Drop process or by creating new key pair using Trust, Ledger or another wallet.

Reviewer: ```suggestion
The owner account is created first, either by following the NEAR Drop process or by creating a new key pair using Trust, Ledger, or another wallet.
```

The lockup contract is then deployed with a predictable name.
It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`.

> It means that all lockup contracts end with `.lockup.near`.

Reviewer: ```suggestion
It means that all lockup contracts are deployed to accounts that are named with `.lockup.near` at the end.
```

> If you have added or earned additional tokens above the initial balance, they are considered unlocked and can be freely withdrawn.

Reviewer: ```suggestion
If you have received additional tokens to the lockup account, they are considered unlocked and can be freely withdrawn.
```

> For example, any rewards that are earned using this lockup contract (eg from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.

Reviewer: ```suggestion
For example, any rewards that are earned using this lockup account (e.g. from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.
```

The actual lockup release process happens on per block basis. 
E.g. if the release length is 1 calendar year, it will actually be `31,536,000` seconds, and with ~1 second blocks, `~1/31,536,000` will be released per block.
When the lockup has been fully released, the Owner Account can add the full-access key and withdraw all the funds from it.

The contract consists of lockup and vesting processes that go simultaneously.
Both of these processes lock the tokens, but the mechanics slightly differ.

### Lockup

Lockup mechanics have 2 configurable parameters: 
1. `lockup_timestamp` - The moment when tokens start linearly unlocking;
2. `release_duration` - The length of the unlocking schedule during which tokens are linearly unlocked.
  By the end of this duration, all tokens are unlocked.

The lockup process could not be terminated.
Lockup does not have a cliff.

<img width="563" alt="Screenshot 2021-05-07 at 14 21 34" src="https://user-images.githubusercontent.com/11246099/117442575-a28ecc80-af3f-11eb-8304-01ce5f2eaa5c.png">

[deprecated] Apart from the lockup timestamp, there is a lockup duration.

> `lockup_duration` is the interval between [the Phase 2 launch](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) (October 14th) and the moment when tokens start to unlock.

Reviewer: ```suggestion
`lockup_duration` is the interval between [the Phase II launch](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) (October 14th) and the moment when tokens start to unlock.
```

![lockup_2](/docs/assets/lockup/lockup_2.png)

### Vesting

Vesting also locks the tokens, and it allows to configure 2 more options:
1. Ability to terminate tokens vesting and refund non-vested tokens back.
2. Cliff vesting period.

> Vesting process contains of 3 timestamps: `start_date`, `cliff_date`, `end_date`.

Reviewer: ```suggestion
Vesting process is defined by 3 timestamps: `start_date`, `cliff_date`, `end_date`.
```

Author: Changed it to `includes` because `is defined by` sounds for me like if I have these 3 timestamps, I should be able to say which vesting process it is.

![lockup_3](/docs/assets/lockup/lockup_3.png)

### Combinations

`v_start`, `v_cliff`, `v_end` are the aliases for vesting parameters; `l_start`, `l_end` are for lockup parameters.
They could be easily transformed into initializing parameters described above.

![lockup_4](/docs/assets/lockup/lockup_4.png)

> The resulting graph of the tokens becoming liquid is always the minimum between lockup and vesting mechanics.

Reviewer: ```suggestion
The liquid tokens balance is always the minimum between unlocked and vested values.
```

### Termination of vesting

Vesting could be terminated by the foundation, an account configured at the moment of initializing the contract.
It's important to understand how the termination works combining with the lockup schedule.

![lockup_5](/docs/assets/lockup/lockup_5.png)

> At the moment of termination, we fix the number of tokens that should be finally unlocked at the end.

Reviewer: ```suggestion
At the moment of termination, we stop the vesting process, so the vested amount is going to remain constant after that; the lockup process keeps going and will unlock the tokens on its schedule.
```

> We continue to unlock the tokens as we normally do that by getting the minimum between lockup and vesting graph.

Reviewer: ```suggestion
We continue to unlock the tokens as we normally do that by getting the minimum between unlocked and vested amounts.
```

> We stop when we reach the desired amount of tokens.

Reviewer: This sentence is unclear :thinking:

Author: Removed it, it does not give any value here
